### PR TITLE
Table: fix #14499, add min-width test for table

### LINKED ISSUE #14499: [Bug Report] Table column min-width and width
### Element UI version
2.5.4

### OS/Browsers version
Mac OS, Chrome

### Vue version
2.6.7

### Reproduction Link
[https://jsfiddle.net/y59dkswv/](https://jsfiddle.net/y59dkswv/)

### Steps to reproduce
Setting table column width and min-width.

### What is Expected?
If min-width is bigger than width; min-width should be applied.

### What is actually happening?
width is applied. min-width is ignored.

### PR DESCRIPTION
**Pull request by yugasun**
Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

- Closes #14499

### CODE REVIEW

File: packages/table/src/layout-observer.js [hunk 1]
```diff
@@ -46,7 +46,10 @@ export default {
         const name = col.getAttribute('name');
         const column = columnsMap[name];
         if (column) {
-          col.setAttribute('width', column.realWidth || column.width);
+          const realWidth = (column.minWidth !== undefined && column.minWidth > column.width)
+            ? column.minWidth
+            : column.width;
```
Comment (SimonaliaChen) [inline packages/table/src/layout-observer.js]: It seems there is no need to check the existense of `column.minWidth`.
I think it's more concise to use Math.max

File: packages/table/src/table-column.js [hunk 2]
```diff
@@ -98,11 +98,15 @@ const getDefaultColumn = function(type, options) {
     }
   }
 
-  if (!column.minWidth) {
+  // if set width no need to set minWidth, because if user want set width less than 80, that will not work.
+  if (!column.minWidth && !column.width) {
```
Comment (SimonaliaChen) [inline packages/table/src/table-column.js]: is it neccessary?

File: packages/table/src/table-column.js [hunk 3]
```diff
@@ -98,11 +98,15 @@ const getDefaultColumn = function(type, options) {
...
     column.minWidth = 80;
   }
 
-  column.realWidth = column.width === undefined ? column.minWidth : column.width;
+  // if minWidth is set ant bigger than width, set realWidth to minWidth
+  column.realWidth = (column.minWidth !== undefined && column.minWidth > column.width)
+    ? column.minWidth
+    : column.width;
```
Comment (SimonaliaChen) [inline packages/table/src/table-column.js]: looks like redundant logic

**Comment by catCarrot** [comment]
It is still not used in version 2.12.0

**Pull request closed without merging**